projects/hathor-wallet-headless/0001-multisig-create-mint-melt-token.md

```diff
@@ -0,0 +1,324 @@
+```ts
+{
+  "success": boolean,
+  "txHex": string
+}
+```
+
+The purpose of the command is only produce the `txHex` not signed and nothing more, it will not push the transaction to the network. The returned `txHex` will be handled by the coordinator service, and later on by the federation service, which is responsible to sign the transaction.
+
+With this specification we cover the possibility to melt tokens, send it to another wallet, create or not another melt authority, send the authority to another wallet, and define an address to deposit the withdraw amount if any.
+
+## Decode transaction with wallet metadata
+
+There is an endpoint to [inspect the transaction](https://hathor.gitbook.io/hathor/guides/multisig-wallet/sending-transactions#5.-check-the-multisig-transaction) by decoding the transaction hexadecimal encode in `POST:/wallet/decode`. However, it doesn't bring signature information, nor wallet information with it.
+
+By extending this endpoint operation to include wallet metadata we introduce a breaking change in the API. However, we keep coherence, once it is used not only to decode a `txHex` but to decode a partial transaction. The extension adds a semantic decoding of the transaction components regarding the wallet. Also, it reduces the roundtrips to the wallet by delivering the tokens balance.
+
+The data field is:
+* txHex
+* partial_tx
+
+The response scheme for success:
+```ts
+{
+  "success": boolean,
+  "complete": boolean,
+  "tx": {
+    "version": number,
+    "tokens": string[],
+    "inputs": [
+      {
+	    txId: string,
+        index: number,
+        decoded: {
+            type: string,
+            address: string,
+            timelock: number,
+        },  // data from the spent ouput
+        token: string,
+        value: number,
+        tokenData: number, // user face
+        token_data: number, // internal use
+        script: string,
+        signed: boolean,
+        mine: boolean,
+      },
+    ],
+    "outputs": [
+      {
+	    decoded: {
+		    address: string,
+		    timelock: number,
+		    mine: boolean,
+	    },
+        value: number,
+        tokenData: number, // user face
+        token_data: number, // internal use
+        script: string,
+        type: string,
+        token?: string,
+      }
+    ],
+  },
+  "balance": {
+    "<token-uid>":  {
+      "tokens": { available: number, locked: number ),
+      "authorities": {
+        mint: { available: number, locked: number ),
+        melt: { available: number, locked: number ),
+      },
+    },
+  },
+}
+```
+
+Lets represent the response document as ` $ `, and each element of a list ` [*] `, we have:
+* `$.complete` -- that represents the completeness of signatures required to use the transaction
+* `$.tx.inputs[*].signed` -- that indicates this input has a signature
+* `$.tx.inputs[*].mine` -- that indicates this input belongs to this wallet
+* `$.tx.outputs[*].decoded.mine` -- that indicates this output belongs to this wallet
```

Comment:
Why is it inside the `decoded` section here but not on the input?

Reply:
Well, it happens because the output can be of type `data` and its structure doesn't have `address`. This way we would have the `mine` attribute with values `undefined` or `null`.
```
export interface IDataOutputData {
  type: 'data';
  token: string;
  value: number;
  authorities: number;
  data: string;
};
```

Reply:
If you see it should be in the body as it happens with the input we can do it. I can add an Appendix to the design, open an issue and implement in another PR. What do you think?

I have opened a bump PR for the wallet-headless version already, but we can do it before the bump. Here is the bump PR: https://github.com/HathorNetwork/hathor-wallet-mobile/pull/304

Reply:
Even when the output is of type data, shouldn't `mine` be `false` as well?

Reply:
This makes sense, we should have the mine on `output.mine` and not on `output.decoded.mine` and if the output is a data output it will be false since a data output does not "belong" to any address.